grafanalib/tests/test_parsing.py

```diff
@@ -0,0 +1,444 @@
+    return st.builds(G.RGB, r=colors(), g=colors(), b=colors())
+
+
+def pixels():
+    """Generate arbitrary valid Pixels object"""
+    return st.builds(G.Pixels, num=st.integers(min_value=0))
+
+
+def percents():
+    return st.builds(G.Percent, num=st.integers(min_value=0, max_value=100))
+
+
+def mappings():
+    return st.builds(G.Mapping, name=st.text(), value=st.integers())
+
+
+def unknown():
+    return st.text(string.printable)
+
+
+def grids():
+    return st.builds(G.Grid,
+                     threshold1=unknown(),
+                     threshold1Color=rgbas(),
+                     threshold2=unknown(),
+                     threshold2Color=rgbas(),
+                     leftLogBase=unknown(),
+                     rightLogBase=unknown(),
+                     rightMin=unknown(),
+                     rightMax=unknown(),
+                     leftMin=unknown(),
+                     leftMax=unknown())
+
+
+def legends():
+    return st.builds(G.Legend,
+                     avg=st.booleans(),
+                     current=st.booleans(),
+                     max=st.booleans(),
+                     min=st.booleans(),
+                     show=st.booleans(),
+                     total=st.booleans(),
+                     values=unknown(),
+                     alignAsTable=st.booleans(),
+                     hideEmpty=st.booleans(),
+                     hideZero=st.booleans(),
+                     rightSide=st.booleans(),
+                     sideWidth=unknown(),
+                     sort=unknown(),
+                     sortDesc=unknown())
+
+
+# def targets():
+#     return st.builds(G.Target,
+#                      # target param is usde by st.builds ;-(
+#                      expr=unknown(),
+#                      legendFormat=st.text(string.printable),
+#                      intervalFacto=st.integers(),
+#                      metric=st.text(string.printable),
+#                      refId=st.text(string.printable),
+#                      step=st.integers(),
+#                      hide=st.booleans(),
+#                      format=unknown(),
+#                      calculatedInterval=unknown(),
+#                      datasourceErrors=unknown(),
+#                      errors=unknown(),
+#                      interval=unknown(),
+#                      # target=unknown(),
+#                      alias=unknown(),
+#                      dimensions=unknown(),
+#                      metricName=unknown(),
+#                      namespace=unknown(),
+#                      period=unknown(),
+#                      region=unknown(),
+#                      statistics=unknown())
+
+
+def tooltips():
+    return st.builds(G.Tooltip,
+                     msResolution=st.booleans(),
+                     shared=st.booleans(),
+                     sort=st.integers(),
+                     valueType=st.text(string.printable))
+
+
+def xaxes():
+    return st.builds(G.XAxis,
+                     # XAxis loses information
```

Comment:
I don't get it. Can you provide more context?

Reply:
There are 4 params (mode, name, values, show) but only 1 (show) is converted to JSON.
https://github.com/weaveworks/grafanalib/blob/master/grafanalib/core.py#L280